lib/dry/types/hash/schema.rb

```diff
@@ -4,15 +4,15 @@ class Hash < Definition
       class Schema < Hash
         attr_reader :member_types
 
-        def initialize(primitive, options = {})
+        def initialize(_primitive, options)
           @member_types = options.fetch(:member_types)
           super
         end
 
         def call(hash, meth = :call)
           member_types.each_with_object({}) do |(key, type), result|
             if hash.key?(key)
-              result[key] = type.__send__(meth, hash[key])
+              result[key] = type.public_send(meth, hash.fetch(key))
```

Comment:
The semantics of `public_send` are more correct in this use case and the performance difference is not statistically significant

```
$ cat public-send.rb
require "benchmark/ips"

class Thing
  def method
  end
end

Benchmark.ips do |x|
  x.report("__send__")    { Thing.new.__send__(:method) }
  x.report("public_send") { Thing.new.public_send(:method)    }
  x.compare!
end

$ ruby public-send.rb
Warming up --------------------------------------
            __send__   259.922k i/100ms
         public_send   242.706k i/100ms
Calculating -------------------------------------
            __send__      5.311M (± 6.4%) i/s -     26.512M in   5.013065s
         public_send      4.854M (± 4.5%) i/s -     24.271M in   5.010566s

Comparison:
            __send__:  5311226.2 i/s
         public_send:  4854437.0 i/s - same-ish: difference falls within error
```